hs_csdms/models.py

```diff
@@ -0,0 +1,69 @@
+from django.db import models, transaction
+from .utils import split_csdms_names
+
+
+class CSDMSName(models.Model):
+    """ Maintain a cached copy of the CSDMS standard name database inside HydroShare
+        This maintains a copy of the CSDMS database maintained at
+        https://csdms.colorado.edu/wiki/CSDMS_Standard_Names
+    """
+    source = models.CharField(max_length=256, editable=False, null=False, blank=False,
+                              help_text='The source file from CSDMS')
+    part = models.CharField(max_length=16, choices=(('name', 'name'), ('decoration', 'decoration'),
+                            ('measure', 'measure')),
+                            help_text='The standard names can be splitted to three parts,\
+                                       name, decoration, and measurement')
+    value = models.CharField(max_length=255, editable=False, null=False, blank=False,
+                             help_text='The value for the corresponding part')
+
+    @staticmethod
+    def add_word(s, p, v):
+        """ add a record to the CSDMS Database
+            :param s, a choice from source describes where the word from
+            :param p, a choice from part describes which part the word is extracted from
+            :param, v, a string value for the word to be added
+        """
+        with transaction.atomic():
```

Comment:
Pretty sure this won't affect get_or_create at all.  remove?

Reply:
removed.